opensir/models/model.py

```diff
@@ -206,29 +212,33 @@ def fit(self, t_obs, n_obs, population, fit_index=None):
         if len(t_obs) != len(n_obs):
             raise self.InconsistentDimensionsError(Exception)
         # if no par_index is provided, fit only the first parameter
-        if (fit_index is None) & (self.fit_index is None):
+        if (fit_index is None) & (self.fit_attr["fit_index"] is None):
```

Comment:
note the "&" is a bitwise and, not a "logical" and. This "&" might produce unexpected results

Reply:
in fact, maybe it's better to drop the second clause. The "fit_index" attribute is set anyway

Reply:
Good point, with the refactor that you suggest it is no longer necessary. This is updated.